tests/migration/test_host_evacuate.py

```diff
@@ -34,6 +34,7 @@ def _host_evacuate_test(source_host, dest_host, network_uuid, vm, expect_error=F
         source_host.xe('host-evacuate', args)
         wait_for(lambda: vm.all_vdis_on_host(dest_host), "Wait for all VDIs on destination host")
         wait_for(lambda: vm.is_running_on_host(dest_host), "Wait for VM to be running on destination host")
+        vm.host = dest_host
```

Comment:
it is not related to wait for xenstore VM IP right? It looks like a bug fix when evacuation is done on the destination host currently self.host is not updated no? If it is the case I think that it is a fix that should be done in its own PR.

Reply:
It is indeed required so the upcoming check for VM IP runs `xenstore` commands on the destination host, although it is a workaround that can likely be used for other things. Could be moved in a separate commit for more clarity.

Reply:
ok I see :+1:

Reply:
perhaps a vm.resident_on would be better.

Reply:
Agreed with @benjamreis here: as I said earlier, vm.host really should be vm.pool and will probably be in the future. It is not where the VM is running, it's the pool master of the pool to which the VM belongs, and is badly named.

You need to use the `VM.get_residence_host` method in `VM.try_get_ip_xenstore`.

Reply:
so we're indeed evacuating the host into another pool?

Reply:
> Agreed with @benjamreis here: as I said earlier, vm.host really should be vm.pool and will probably be in the future. It is not where the VM is running, it's the pool master of the pool to which the VM belongs, and is badly named.

And in the test we are sure that we migrate a VM from a slave to the master (or another slave) during the evacuate? Because if the evacuation is done from the master than we need to update vm.host in this case. Or we are sure that it cannot happens.

Reply:
Oh wait, in fact no because evacuate from master doesn't mean that we need to change the master... maybe.

Reply:
What I mean is VM.host is not where the VM is currently executed. There is no guarantee about this, and it usually just is the master host, regardless of where the VM is (note also that the concept of residence for a VM on a host doesn't work for a VM which is turned off and is on a shared SR. When you start it from XAPI, any host of the pool can become temporarily where it resides). And thus we shouldn't change the value in the test, and use `VM.get_residence_host` in order to find which host's xenstore we need to check.

> so we're indeed evacuating the host into another pool?

No, host evacuate is intra-pool and doesn't involve storage migration.